### PR TITLE
PlayStack:Apply PlayStackManager in Capability Agents

### PR DESCRIPTION
**Pull request by kimhyungrok**
It update PlayStackManager which is possible to
process context management policy.

It remove PlaySyncManager and related codes.

It apply PlayStackManager in ASRAgent,
TTSAgent, AudioPlayerAgent, TextAgent.

It add playStack field in context by default.

Signed-off-by: kimhyungrok <hr97gdi@gmail.com>

### CODE REVIEW

File: src/core/playstack_manager.cc [hunk 1]
```diff
@@ -21,6 +21,50 @@
...
+    {
+        std::string playstack_log("\n>>>>> [PlayStack] >>>>>\n");
+
+        for (const auto& item : playstack_container.second) {
+            std::string layer = "[None] ";
+
+            try {
+                switch (playstack_container.first.at(item)) {
+                case PlayStackLayer::Alert:
+                    layer = "[Albert] ";
```
Comment (webispy) [inline src/core/playstack_manager.cc:38]: Please fix the typo "Al**b**ert"

File: src/core/playstack_manager.cc [hunk 2]
```diff
@@ -21,6 +21,50 @@
...
+                    break;
+                case PlayStackLayer::Call:
+                    layer = "[Call] ;";
```
Comment (webispy) [inline src/core/playstack_manager.cc:41]: Please remove a trailing '**;**'

File: src/core/playstack_manager.cc (hunk 1 above)
Comment (kimhyungrok) [reply src/core/playstack_manager.cc:38]: I'll fix it. Thanks.

File: src/core/playstack_manager.cc (hunk 2 above)
Comment (kimhyungrok) [reply src/core/playstack_manager.cc:41]: I'll fix it. Thanks.

File: src/capability/audio_player_agent.cc [hunk 3]
```diff
@@ -985,15 +981,11 @@ void AudioPlayerAgent::parsingStop(const char* message)
...
-        std::string stacked_ps_id = playsync_manager->getPlayStackItem(getName());
+        playstack_manager->remove(playstackctl_ps_id, is_finished ? PlayStackRemoveMode::Normal : PlayStackRemoveMode::Immediately);
 
         capa_helper->sendCommand("AudioPlayer", "ASR", "releaseFocus", "");
 
-        if (!stacked_ps_id.empty() && playstackctl_ps_id != stacked_ps_id)
-            return;
-
-        if (stacked_ps_id.empty())
+        if (!is_finished)
```
Comment (JeanTracker) [inline src/capability/audio_player_agent.cc:988]: If the audioplayer doesn't play anymore, the audio player release the focus itself.
I think the condition `if (!is_finished)` should be removed.

File: src/capability/tts_agent.cc [hunk 4]
```diff
@@ -406,19 +409,20 @@ void TTSAgent::parsingSpeak(const char* message)
...
-
     cur_token = token;
     ps_id = play_service_id;
 
     is_finished = false;
     speak_dir = getNuguDirective();
     dialog_id = nugu_directive_peek_dialog_id(speak_dir);
 
+    std::string tmp_playstackctl_ps_id = getPlayServiceIdInStackControl(root["playStackControl"]);
+    playstack_manager->add(tmp_playstackctl_ps_id, speak_dir);
```
Comment (JeanTracker) [inline src/capability/tts_agent.cc:420]: Is it okay to add `tmp_playstackctl_ps_id` whether empty value or not?

File: src/capability/audio_player_agent.cc (hunk 3 above)
Comment (kimhyungrok) [reply src/capability/audio_player_agent.cc:988]: I agree. The previous code is not reasonable, even though I converted it to current code as maintaining logic.
I'll remove it.

File: src/capability/tts_agent.cc (hunk 4 above)
Comment (kimhyungrok) [reply src/capability/tts_agent.cc:420]: The add method of PlayStackManager check the validation about value, so, it's no problem.

In addition, it needs to pass "tmp_playstackctl_ps_id" directly to PlayStackManager, even if it's empty.

**Pull request merged**